### PR TITLE
Updated ssh fingerprint hash-type default to sah256 for Nitrogen and remove md5 warnings

### PR DESCRIPTION
**Pull request by rallytime**
Refs #40543

### CODE REVIEW

**Review by cachedout — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

File: salt/modules/ssh.py [hunk 1]
```diff
@@ -237,23 +237,16 @@ def _fingerprint(public_key, fingerprint_hash_type=None):
...
 
     '''
     if fingerprint_hash_type:
         hash_type = fingerprint_hash_type.lower()
     else:
-        # Set fingerprint_hash_type to md5 as default
-        log.warning('Public Key hashing currently defaults to "md5". This will '
-                    'change to "sha256" in the Nitrogen release.')
-        hash_type = 'md5'
+        hash_type = 'sha256'
```
Comment (cachedout) [inline salt/modules/ssh.py]: Looks like this is breaking the tests.

**Review by cachedout — CHANGES_REQUESTED** [review]

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
@cachedout These failing tests should be fixed now. :)

**Review by cachedout — APPROVED** [review]

**Comment by rallytime** [comment]
This is causing the tests to hang on the nitrogen branch right here:
```
17:12:57 Starting External Pillar Tests
17:12:57 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
17:12:57 
17:12:57 Running tests...
17:12:57 ----------------------------------------------------------------------
17:12:57   test_includes_disabled (integration.pillar.test_git_pillar.TestGitPythonAuthenticatedHTTP)
17:12:57 [CPU:100.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_includes_enabled (integration.pillar.test_git_pillar.TestGitPythonAuthenticatedHTTP)
17:12:57 [CPU:0.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_multiple_sources_dev_master_merge_lists (integration.pillar.test_git_pillar.TestGitPythonAuthenticatedHTTP)
17:12:57 [CPU:0.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_multiple_sources_dev_master_no_merge_lists (integration.pillar.test_git_pillar.TestGitPythonAuthenticatedHTTP)
17:12:57 [CPU:0.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_multiple_sources_master_dev_merge_lists (integration.pillar.test_git_pillar.TestGitPythonAuthenticatedHTTP)
17:12:57 [CPU:0.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_multiple_sources_master_dev_no_merge_lists (integration.pillar.test_git_pillar.TestGitPythonAuthenticatedHTTP)
17:12:57 [CPU:0.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_multiple_sources_with_pillarenv (integration.pillar.test_git_pillar.TestGitPythonAuthenticatedHTTP)
17:12:57 [CPU:0.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_single_source (integration.pillar.test_git_pillar.TestGitPythonAuthenticatedHTTP)
17:12:57 [CPU:0.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_includes_disabled (integration.pillar.test_git_pillar.TestGitPythonHTTP)
17:12:57 [CPU:100.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_includes_enabled (integration.pillar.test_git_pillar.TestGitPythonHTTP)
17:12:57 [CPU:0.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_multiple_sources_dev_master_merge_lists (integration.pillar.test_git_pillar.TestGitPythonHTTP)
17:12:57 [CPU:0.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_multiple_sources_dev_master_no_merge_lists (integration.pillar.test_git_pillar.TestGitPythonHTTP)
17:12:57 [CPU:0.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_multiple_sources_master_dev_merge_lists (integration.pillar.test_git_pillar.TestGitPythonHTTP)
17:12:57 [CPU:0.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_multiple_sources_master_dev_no_merge_lists (integration.pillar.test_git_pillar.TestGitPythonHTTP)
17:12:57 [CPU:0.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_multiple_sources_with_pillarenv (integration.pillar.test_git_pillar.TestGitPythonHTTP)
17:12:57 [CPU:100.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_single_source (integration.pillar.test_git_pillar.TestGitPythonHTTP)
17:12:57 [CPU:0.0%|MEM:73.7%]  ... SKIP (0.001s)
17:12:57   test_includes_disabled (integration.pillar.test_git_pillar.TestGitPythonSSH)
17:12:59 [CPU:100.0%|MEM:73.7%]  ... ERROR (2.482s)
17:12:59   test_includes_enabled (integration.pillar.test_git_pillar.TestGitPythonSSH)
17:13:04 The authenticity of host '[127.0.0.1]:39093 ([127.0.0.1]:39093)' can't be established.
17:13:04 RSA key fingerprint is SHA256:vZmsqXbFBKUmMD6eCLItW5rl5xGGWI/eB/EkXpkAmJc.
```
So I am reverting this for now. I'll have to resubmit once those tests are fixed.

**Pull request merged**